### DISCUSSION TITLE
Best practices or how-to's

### DISCUSSION THREAD

**Discussion by scherer-michael** [gabime/spdlog#2873] [Q&A]
Hi community 🖐️
I am using spdlog for a few month and I was wondering if it existed a "best practice book" or "How-to" to improve my knowledge in using the library.
Currently, I'm only using few of the existing sinks and the default logger, created by the library. I know that I am missing lots of interesting features of the library, but I don't really know which use case they are answering and/or if my situations really look like those use cases.

As a simple example, I was wondering if it was good practice, with spdlog, to create various loggers for various modules, and to do it like Python, for example : `logger = logging.getLogger(__name__)` (in a new Python module) ?
It would, then, create a global variable, which is not considered "good practice" and make testing more difficult.

My question : Does a document / book / blog post exist, which explains a bit more how to use this wonderful library ? If not, do you think that it would be a good idea to have one ?

(please, don't get me wrong : the already-existing presentation is super useful to know what is inside the library and the work behind all these features is incredible — really thank you for this library —, but I just would want to find something, like a Cook book or else, to learn how to use it at its 100% and not only the defaults).

Thank you all for your advice and/or ideas!
Regards,
MSC

**Comment by gabime** [comment]
Hi @scherer-michael,

Thanks for the nice feedback 👍🏼 
There is no how-to book, however there is the [wiki](https://github.com/gabime/spdlog/wiki) with useful information and a FAQ. It is free for everyone to edit. Also, it is worthwhile to browse past issues.

Hope it helps,
Gabi
